builder/vmware/iso/step_create_vmx.go

```diff
@@ -396,23 +399,28 @@ func (s *stepCreateVMX) Run(_ context.Context, state multistep.StateBag) multist
 
 	/// Use the disk adapter type that the user specified to tweak the .vmx
 	//  Also sync the cdrom adapter type according to what's common for that disk type.
+	//  XXX: If the cdrom type is modified, make sure to update common/step_clean_vmx.go
```

Comment:
I don't think a comment is going to have the enforcement we're looking for. any way to turn this into a test?

Reply:
Afaict, turning it into a test will still need some kind of enforcement as if a new device type is added (not likely, but still possible) both the test itself and the steps too will still need knowledge of what types are available in order to guarantee detaching the iso.

This also becomes awkward as a user can also affect the type out-of-band using the `vmx_data` option which can bypass these checks entirely. This is why I assume the author chose to use a regex to select and detach all iso types (that packer knows about) which should cover all the cases where the step_clean_vmx.go step might not detach the iso.

This would be necessary unless someone refactors the device type selection to expose the available types (for disk and cdrom) as some sort of data instead of using if-tests/string-comparisons and regexes like the rest of the builders do (parallels comes to mind as doing this too).

In summary, the coupling between steps that attach and detach the iso as a result of packer needing knowledge of what device types a user might choose will still require someone to synchronize the the type checks in more than one place whether it's in a test or a step.

Reply:
I'm in favor of refactoring this selection to use some kind of data structure and actually fixing the problem, rather than continuing to throw band-aids onto it.